### PR TITLE
fix: fix transitioning issue from old dot

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/27418
PROPOSAL: https://github.com/Expensify/App/issues/27418#issuecomment-1746570494

### Tests

1. Log in to expensify.com with one of your accounts (account A).
2. Log in to the new app with another account (account B).
3. On expensify.com go to Support and choose Concierge.
4. A new tab will open, when a long URL with transition and token appear, quickly copy it. *
5. Open a new tab (or in a tab with the app running) paste the address and change 'https://new.expensify.com' for 'http://localhost:8082' [or any port your app is running] *
6. Verify that user is properly logged in to the account A and the sidebar with chat and main screen are not empty.

* steps required for testing locally

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

1. Log in to expensify.com with one of your accounts (account A).
2. Log in to the new app with another account (account B).
3. On expensify.com go to Support and choose Concierge.
4. A new tab will open, when a long URL with transition and token appear, quickly copy it. *
5. Open a new tab (or in a tab with the app running) paste the address and change 'https://new.expensify.com' for 'http://localhost:8082' [or any port your app is running] *
6. Verify that user is properly logged in to the account A and the sidebar with chat and main screen are not empty.

* steps required for testing locally

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/28020445/de0a951d-eb2a-4621-94cc-5a7d8fb2f3d7

</details>

<details>
<summary>Mobile Web - Chrome</summary>

N/A
</details>

<details>
<summary>Mobile Web - Safari</summary>

N/A
</details>

<details>
<summary>Desktop</summary>

N/A
</details>

<details>
<summary>iOS</summary>

N/A
</details>

<details>
<summary>Android</summary>

N/A
</details>

### CODE REVIEW

File: src/pages/LogOutPreviousUserPage.js [hunk 1]
```diff
@@ -25,31 +23,18 @@ const defaultProps = {
...
-                const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL, sessionEmail);
-
-                if (isLoggingInAsNewUser) {
-                    Session.signOutAndRedirectToSignIn();
-                }
-
-                // We need to signin and fetch a new authToken, if a user was already authenticated in NewDot, and was redirected to OldDot
-                // and their authToken stored in Onyx becomes invalid.
-                // This workflow is triggered while setting up VBBA. User is redirected from NewDot to OldDot to set up 2FA, and then redirected back to NewDot
-                // On Enabling 2FA, authToken stored in Onyx becomes expired and hence we need to fetch new authToken
```
Comment (BeeMargarida) [inline src/pages/LogOutPreviousUserPage.js:39]: I think this comment is still valid, no?

Comment (koko57) [reply src/pages/LogOutPreviousUserPage.js:39]: ah, yes! thanks

File: src/libs/Navigation/AppNavigator/AuthScreens.js [hunk 2]
```diff
@@ -154,7 +166,7 @@ class AuthScreens extends React.Component {
             User.subscribeToUserEvents();
         });
 
-        // If we are on this screen then we are "logged in", but the user might not have "just logged in". They could be reopening the app
+        // // If we are on this screen then we are "logged in", but the user might not have "just logged in". They could be reopening the app
```
Comment (BeeMargarida) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:169]: small fix
```suggestion
        // If we are on this screen then we are "logged in", but the user might not have "just logged in". They could be reopening the app
```

**Review by BeeMargarida — APPROVED** [review]

**Review by adhorodyski — APPROVED** [review]

**Comment by koko57** [comment]
@allroundexperts it didn't happen before the changes?

**Comment by koko57** [comment]
@allroundexperts I've checked it - it also happened before on main. So my changes didn't introduce the issue.
https://github.com/Expensify/App/assets/28020445/1f9eb244-e506-4e24-bf09-96e83881d87a

@mountiny @allroundexperts I'm not sure if this should be fixed here or should be done as a follow-up issue?

**Comment by koko57** [comment]
@allroundexperts but I think we have to wait for this one https://github.com/Expensify/App/issues/26530

**Comment by koko57** [comment]
@allroundexperts The issue and the solution proposed in this issue are pretty similar to the one that I proposed. I just found this issue accidentally after I made a PR here. But the PR from this other issue was made earlier. My approach is slightly different though, and we also need to check if it solves this other problem, although I think it does and testing locally I works fine for me.

**Comment by mountiny** [comment]
I think the tab crashing is on main yeah

**Comment by burczu** [comment]
@vit @allroundexperts I came from https://github.com/Expensify/App/issues/26530 - In my opinion the solution implemented by this PR fixes both issues, the only concern that @sourcecodedeveloper (author of the other solution) has, is that it breaks deeplinking, but I'm not sure it's the case here. In fact, with the solution from this PR, if we paste the `/transition&...` link in the chat and then click it the app stuck showing spinner, but I can't believe it's a real life scenario and I'm not sure if we wan't to handle such cases. What do you think?

**Comment by mountiny** [comment]
@allroundexperts could we resume the checklist on this one please

**Comment by koko57** [comment]
@allroundexperts @mountiny I'll merge with the newest main

**Comment by sourcecodedeveloper** [comment]
@mountiny The current PR will not be resolving the issue of loading spinner as explained earlier. What I would suggest is the PR https://github.com/Expensify/App/pull/28819 be merged first, which would then take care of the loading spinner and the workspace issue.

**Comment by koko57** [comment]
@MonilBhavsar could you please tell me a little bit more what's happening there? What are the exact steps to reproduce to get a loading spinner?

**Comment by NikkiWines** [comment]
> @MonilBhavsar could you please tell me a little bit more what's happening there? What are the exact steps to reproduce to get a loading spinner?

Not @MonilBhavsar, but the issue is reproducible by logging into expensify.com and clicking the green chat bubble icon on the `Inbox` page, which redirects you to New Expensify.

**Comment by koko57** [comment]
@NikkiWines Thanks for the response! Yeah, looks fixed now 

https://github.com/Expensify/App/assets/28020445/e39fa48d-6301-46bd-b7e3-3c7f024c874a

**Comment by koko57** [comment]
@allroundexperts I think it's ready for re-review 🙂

File: src/libs/Navigation/AppNavigator/AuthScreens.js [hunk 3]
```diff
@@ -145,6 +146,16 @@ class AuthScreens extends React.Component {
     }
 
     componentDidMount() {
+        const currentUrl = getCurrentUrl();
+
+        const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(currentUrl, this.props.session.email);
```
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:151]:
```suggestion
        const currentUrl = getCurrentUrl();
        const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(currentUrl, this.props.session.email);

```

**Review by mountiny — DISMISSED** [review]
One NAB @koko57

File: src/libs/Navigation/AppNavigator/AuthScreens.js (hunk 3 above)
Comment (koko57) [reply src/libs/Navigation/AppNavigator/AuthScreens.js]: @mountiny done 😃

**Review by mountiny — APPROVED** [review]
THis works for me locally with my VM but also noticed that it crashes in the other tab. I dont think that is a blocker but I swear we had fix for this in place
https://github.com/Expensify/App/assets/36083550/05210e1e-042d-4385-8e4e-35c114c3a231

**Comment by mountiny** [comment]
Updated the checklist from @allroundexperts and moving this ahead

**Comment by koko57** [comment]
@mountiny on these crashes - Ana was working on a quite similar issue https://github.com/Expensify/App/issues/15321 I think that her PR may solve this issue https://github.com/Expensify/App/issues/15321

**Comment by mountiny** [comment]
This PR right? https://github.com/Expensify/react-native-onyx/pull/382 anyone able to take it over?

**Comment by koko57** [comment]
I can take it over, or someone from our team can 🙂

**Pull request merged**